config-ui/src/hooks/data-scope/useTransformationsManager.jsx

```diff
@@ -0,0 +1,236 @@
+      break
+    case Providers.JENKINS:
+      transforms = {
+        productionPattern: '',
+        deploymentPattern: ''
+        // stagingPattern: '',
+        // testingPattern: ''
+      }
+      break
+    case Providers.GITLAB:
+      transforms = {
+        productionPattern: '',
+        deploymentPattern: ''
+        // stagingPattern: '',
+        // testingPattern: ''
+      }
+      break
+    case Providers.TAPD:
+      // @todo: complete tapd transforms #2673
+      transforms = {
+        issueTypeRequirement: '',
+        issueTypeBug: '',
+        issueTypeIncident: '',
+        productionPattern: '',
+        deploymentPattern: ''
+        // stagingPattern: '',
+        // testingPattern: ''
+      }
+      break
+  }
+  return transforms
+}
+
+// manage transformations in one place
+const useTransformationsManager = () => {
+  const [transformations, setTransformations] = useState({})
+
+  const generateKey = (
+    connectionProvider,
+    connectionId,
+    projectNameOrBoard
+  ) => {
+    let key = `not-distinguished`
+    switch (connectionProvider) {
+      case Providers.GITHUB:
+      case Providers.GITLAB:
+      case Providers.JENKINS:
+        key = projectNameOrBoard?.id
+        break
+      case Providers.JIRA:
+        key = projectNameOrBoard?.id
+        break
+    }
+    return `${connectionProvider}/${connectionId}/${key}`
+  }
+
+  // change some setting in specific connection's specific transformation
+  const changeTransformationSettings = useCallback(
+    (connectionProvider, connectionId, projectNameOrBoard, settings) => {
+      const key = generateKey(
+        connectionProvider,
+        connectionId,
+        projectNameOrBoard
+      )
+      console.info(
+        '>> SETTING TRANSFORMATION SETTINGS PROJECT/BOARD...',
+        key,
+        settings
+      )
+      setTransformations((existingTransformations) => ({
+        ...existingTransformations,
+        [key]: new TransformationSettings({
+          ...existingTransformations[key],
+          ...settings
+        })
+      }))
+    },
+    [setTransformations]
+  )
+
+  // set a default value for connection's specific transformation
+  const initializeDefaultTransformation = useCallback(
+    (connectionProvider, connectionId, projectNameOrBoard) => {
+      const key = generateKey(
+        connectionProvider,
+        connectionId,
+        projectNameOrBoard
+      )
+      console.info(
+        '>> INIT DEFAULT TRANSFORMATION SETTINGS PROJECT/BOARD...',
+        key
+      )
+      if (!transformations[key]) {
+        setTransformations((old) => ({
+          ...old,
+          [key]: new TransformationSettings(
+            getDefaultTransformations(connectionProvider)
+          )
+        }))
+      }
+    },
+    [setTransformations, transformations]
+  )
+
+  // get specific connection's specific transformation
+  const getTransformation = useCallback(
+    (connectionProvider, connectionId, projectNameOrBoard) => {
```

Comment:
For accuracy with existing design and future compatibility for all entity types, I think this var `projectNameOrBoard` should be refactored to `entityId`

```suggestion
    (connectionProvider, connectionId, entityId) => {
```

Reply:
ok, `projectNameOrBoard` => `entityId` or `entity` will change on next PR.